modules/gapi/include/opencv2/gapi/streaming/onevpl/accel_types.hpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ACCEL_TYPES_HPP
+#define GAPI_STREAMING_ONEVPL_ACCEL_TYPES_HPP
+
+#include <limits>
+#include <string>
+
+#include "opencv2/gapi/own/exports.hpp" // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+enum class AccelType: uint8_t {
+    HOST,
+    DX11,
+
+    LAST_VALUE = std::numeric_limits<uint8_t>::max()
+};
+
+GAPI_EXPORTS const char* to_cstring(AccelType type);
+
+struct IDeviceSelector;
+struct GAPI_EXPORTS Device {
+    friend struct IDeviceSelector;
+    using Ptr = void*;
+
+    ~Device();
+    const std::string& get_name() const;
+    Ptr get_ptr() const;
+    AccelType get_type() const;
+private:
+    Device(Ptr device_ptr, const std::string& device_name,
+           AccelType device_type);
+
+    std::string name;
+    Ptr ptr;
+    AccelType type;
+};
+
+struct GAPI_EXPORTS Context {
+    friend struct IDeviceSelector;
+    using Ptr = void*;
+
+    ~Context();
+    Ptr get_ptr() const;
+    AccelType get_type() const;
+private:
+    Context(Ptr ctx_ptr, AccelType ctx_type);
+    Ptr ptr;
+    AccelType type;
+};
```

Comment:
`gapi::onevpl::Device` & `gapi::onevpl::Context` were designed in type-erased way just to carry non-typed device pointer and context to data consumers. They are not related to vpl, just declared in its namespace. From my perspective any 'Priv' and interface like solution are superfluous here. Instead that we should move Device/Context out from 'cv::gapi::wip::onevpl' and put it into `cv::gapi::wip`
Additionally renaming would required as @TolyaTalamanov  mentioned in one of his comment:
`face_net.cfgPreprocessingParams(...)` on `face_net.cfgVPPPreprocessingParams(...)` because now `vpp` intention in preprocessing creation is expressing by namespace `onevpl` for which Device/Context belong now

Reply:
Let's apply these changes in next PR under the same opencv release edition